pkg/controller/atlascluster/cluster.go

```diff
@@ -0,0 +1,59 @@
+package atlascluster
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlas"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func ensureClusterState(ctx context.Context, wctx *workflow.Context, connection atlas.Connection, project *mdbv1.AtlasProject, cluster *mdbv1.AtlasCluster) (c *mongodbatlas.Cluster, _ workflow.Result) {
+	c = &mongodbatlas.Cluster{}
+
+	client, err := atlas.Client(connection, wctx.Log)
+	if err != nil {
+		wctx.Log.Errorf("Failed to read Atlas Connection details: %s", err.Error())
```

Comment:
considering that `statushandler.Update` prints the last condition happened I believe this (and the other loggings in this file) may produce redundant messages - do you think we can omit these for error/pending cases?

Reply:
True - will remove